packages/@sfcc-core/core/src/index.ts

```diff
@@ -125,6 +143,14 @@ class Core {
             'Create the Core instance used to register services and extensions.',
         );
     }
+
+    init() {
+        this.logger.log('Reinitialized core');
+        this._services = {};
+        this._extensions = {};
+        this._factoryServices = {};
+        this._factoryExtensions = {};
```

Comment:
I added this `init()` method for tests so that I can reinitialize the class internals on each test